### PR TITLE
Add a possibility to push a statement to a transaction

### PR DESCRIPTION
**Pull request by matthiasbeyer**
Targets #284

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@neithernut I'm ready here, do you want to add more functionality to the transaction types? (I would welcome more features)

File: src/objects/message/transaction.c [hunk 1]
```diff
@@ -141,6 +142,42 @@ ws_transaction_commands(
...
+    if (!t->cmds) {
+        t->cmds = calloc(1, sizeof(*t->cmds));
+        if (!t->cmds) {
+            return -ENOMEM;
+        }
+
+        t->cmds->statements = NULL;
+        t->cmds->n          = 0;
+    } else {
+        if (t->cmds->n - 1 == t->cmds->next) {
```
Comment (neithernut) [inline src/objects/message/transaction.c]: Will break if `t->cmds->n == 0`. Do `t->cmds->next +1 >= t->cmds->n` instead.

File: src/objects/message/transaction.c [hunk 2]
```diff
@@ -141,6 +142,42 @@ ws_transaction_commands(
...
+            size_t newsize = (t->cmds->n * 2) * sizeof(*t->cmds->statements);
+
+            tmp = realloc(t->cmds->statements, newsize);
+
+            if (!tmp) {
+                return -ENOMEM;
+            }
+
+            t->cmds->statements = tmp;
+            t->cmds->n *= 2;
```
Comment (neithernut) [inline src/objects/message/transaction.c:170]: You can simply assign `newsize`.

Comment (matthiasbeyer) [inline src/objects/message/transaction.c:170]: Nope, not if `*t->cmds->statements` has more than one byte.

Comment (neithernut) [inline src/objects/message/transaction.c:170]: Ahhh, yep.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**